addons/base_vat/models/res_partner.py

```diff
@@ -222,7 +222,17 @@ def _build_vat_error_message(self, country_code, wrong_vat, record_label):
         else:
             company = self.env.company
 
-        vat_label = company.country_id.vat_label if company.country_id else "Tax ID"
+        vat_label = _("VAT")
+        if country_code:
+            partner_country_data = self.env["res.country"].search_read(domain=[("code", "=", country_code.upper())], fields=["vat_label"], limit=1)
+            partner_vat_label = partner_country_data and partner_country_data[0]["vat_label"] or False
+            # If the partner and company both has same country vat label, vat_label only
+            if partner_vat_label and partner_vat_label == company.country_id.vat_label:
+                vat_label = partner_vat_label
+            # If the partner and company both has different country vat label, VAT with vat_label like "VAT (vat_label)"
+            elif partner_vat_label and partner_vat_label != vat_label:
+                vat_label += "(" + partner_vat_label + ")"
+
```

Comment:
```suggestion
        vat_label = _("VAT")
        if country_code and company.country_id and country_code == company.country_id.code.lower():
            vat_label = company.country_id.vat_label
```

Now if country_code is the same as company country_code then we change the vat label else it's default 'VAT'